src/newTest/java/org/opensearch/test/framework/cluster/LocalOpenSearchCluster.java

```diff
@@ -0,0 +1,610 @@
+        }
+
+        public PluginAwareNode esNode() {
+            return node;
+        }
+
+        public boolean isRunning() {
+            return running;
+        }
+
+        public <X> X getInjectable(Class<X> clazz) {
+            return node.injector().getInstance(clazz);
+        }
+
+        public void stop() {
+            try {
+                log.info("Stopping {}", this);
+
+                running = false;
+
+                if (node != null) {
+                    node.close();
+                    node = null;
+                    Thread.sleep(10);
+                }
+
+            } catch (Throwable e) {
+                log.warn("Error while stopping " + this, e);
+            }
+        }
+
+        @Override
+        public String toString() {
+            String state = running ? "RUNNING" : node != null ? "INITIALIZING" : "STOPPED";
+
+            return nodeName + " " + state + " [" + transportPort + ", " + httpPort + "]";
+        }
+
+        public boolean isPortCollision() {
+            return portCollision;
+        }
+
+        public String getNodeName() {
+            return nodeName;
+        }
+
+        @Override
+        public InetSocketAddress getHttpAddress() {
+            return httpAddress;
+        }
+
+        @Override
+        public InetSocketAddress getTransportAddress() {
+            return transportAddress;
+        }
+
+        private Settings getEsSettings() {
+            Settings settings = getMinimalEsSettings();
+
+            if (nodeSettingsSupplier != null) {
+                // TODO node number
+                return Settings.builder().put(settings).put(nodeSettingsSupplier.get(0)).build();
+            }
+
+            return settings;
+        }
+
+        private Settings getMinimalEsSettings() {
+            return Settings.builder().put("node.name", nodeName).put("node.data", nodeSettings.dataNode).put("node.master", nodeSettings.masterNode)
+                    .put("cluster.name", clusterName).put("path.home", nodeHomeDir.toPath()).put("path.data", dataDir.toPath())
+                    .put("path.logs", logsDir.toPath()).putList("cluster.initial_master_nodes", initialMasterHosts)
+                    .put("discovery.initial_state_timeout", "8s").putList("discovery.seed_hosts", seedHosts).put("transport.tcp.port", transportPort)
+                    .put("http.port", httpPort).put("cluster.routing.allocation.disk.threshold_enabled", false)
+                    .put("discovery.probe.connect_timeout", "10s").put("discovery.probe.handshake_timeout", "10s").put("http.cors.enabled", true)
+                    .build();
+        }
+
+        @Override
+        public String getClusterName() {
+            return clusterName;
+        }
+
+        @Override
+        public TestCertificates getTestCertificates() {
+            return testCertificates;
+        }
+    }
+    
+    private static int getUnitTestForkNumber() {
+        String forkno = System.getProperty("forkno");
```

Comment:
If this value isn't set aren't bad things going to happen during port allocation?  What do you think about fast-failing here?

Reply:
Done: The system property `forkno` was never set, therefore the method was removed